File: financial-domain-starter/src/main/java/com/nixsolutions/financial/security/SecurityProperties.java
package com.nixsolutions.financial.security;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;
import lombok.Getter;
import lombok.Setter;

@Component
@ConfigurationProperties(prefix = "financial-domain.security")
@Getter
@Setter
public class SecurityProperties
{
  private String jwtSecret;

  private Integer expiresIn;

  private Integer passwordHashingStrength;
}

File: financial-domain-starter/src/main/kotlin/com/nixsolutions/financial/security/SecurityProperties.kt
package com.nixsolutions.financial.security

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties(prefix = "financial-domain.security")
@ConditionalOnProperty("financial-domain.security.jwtSecret")
class SecurityProperties {
  var jwtSecret: String? = null

  var expiresIn: Int? = null

  var passwordHashingStrength: Int? = null
}
